app/models/category_list.rb

```diff
@@ -135,21 +135,21 @@ def dismissed_topic?(topic)
   end
 
   def find_categories
+    # Enforce paginaion for users who can see a large number of categories to
+    # smooth out the performance of the category list page.
+    paginate =
+      Category.secured(@guardian).count > MAX_UNOPTIMIZED_CATEGORIES ||
+        @guardian.can_lazy_load_categories?
+
     query = Category.includes(CategoryList.included_associations).secured(@guardian)
     query = self.class.order_categories(query)
 
-    if @options[:parent_category_id].present? || @guardian.can_lazy_load_categories?
+    if @options[:parent_category_id].present? || paginate
       query = query.where(parent_category_id: @options[:parent_category_id])
     end
 
-    style =
-      if Category.secured(@guardian).count > MAX_UNOPTIMIZED_CATEGORIES
-        "categories_only_optimized"
-      else
-        SiteSetting.desktop_category_page_style
-      end
     page = [1, @options[:page].to_i].max
-    if style == "categories_only_optimized" || @guardian.can_lazy_load_categories?
```

Comment:
Looks like we also have to remove https://github.com/discourse/discourse/blob/bba9a99e2e9127c90c0b5f6e983692c0e2c5a493/app/models/category_page_style.rb#L13 too.

Reply:
Oops, I missed that one.